File: src/python/telegram/telegram_manager.py
"""
Module: telegram_manager.py

This module provides a class for sending messages to a Telegram group using the python-telegram-bot library and logging with Loguru.

Classes:
    TelegramManager: A class for managing a Telegram bot to send messages to a group.

Dependencies:
    python-telegram-bot: A wrapper for the Telegram Bot API.
    loguru: A library for logging in Python with simplicity and power.

Usage Example:
    >>> from telegram_manager import TelegramManager
    >>> import asyncio

    >>> # Initialize a TelegramManager object
    >>> token = 'YOUR_TELEGRAM_BOT_TOKEN'
    >>> chat_id = 'YOUR_CHAT_ID'
    >>> manager = TelegramManager(token, chat_id)

    >>> # Send a message to the group
    >>> message = 'Hello, Telegram Group!'
    >>> asyncio.run(manager.send_message_to_group(message))
"""

from telegram import Bot
from telegram.error import TelegramError
from loguru import logger


class TelegramManager:
    """
    A class for managing a Telegram bot to send messages to a group.
    """
    def __init__(self, token: str, chat_id: str) -> None:
        """
        Initializes the TelegramManager with the provided token and chat ID.

        Args:
            token (str): The token of the Telegram bot.
            chat_id (str): The chat ID of the Telegram group.
        """
        self.token = token
        self.chat_id = chat_id
        self.bot = Bot(token=self.token)

        logger.add("file.log", rotation="1 MB")

    async def send_message_to_group(self, message: str) -> None:
        """
        Sends a message to the Telegram group.

        Args:
            message (str): The message to be sent to the group.

        Raises:
            TelegramError: If there is an error sending the message.
        """
        try:
            logger.info(f"Sending message to chat_id {self.chat_id}")
            await self.bot.send_message(chat_id=self.chat_id, text=message)
            logger.info(f"Message sent: {message}")
        except TelegramError as e:
            logger.error(f"Failed to send message: {e}")

File: src/javascript/telegram/telegramManager.js
/**
 * Module: telegramManager.js
 *
 * This module provides a class for sending messages to a Telegram group using the node-telegram-bot-api library.
 *
 * Classes:
 *     TelegramManager: A class for managing a Telegram bot to send messages to a group.
 *
 * Dependencies:
 *     node-telegram-bot-api: A library for interacting with the Telegram Bot API.
 *
 * Usage Example:
 *     const TelegramManager = require('./telegram_manager');
 *
 *     // Initialize a TelegramManager object
 *     const token = 'YOUR_TELEGRAM_BOT_TOKEN';
 *     const chatId = 'YOUR_CHAT_ID';
 *     const manager = new TelegramManager(token, chatId);
 *
 *     // Send a message to the group
 *     const message = 'Hello, Telegram Group!';
 *     manager.sendMessageToGroup(message)
 *         .then(response => console.log('Message sent successfully:', response))
 *         .catch(error => console.error('Error sending message:', error));
 */

const TelegramBot = require("node-telegram-bot-api");

class TelegramManager {
  /**
   * Initializes the TelegramManager with the provided token and chat ID.
   *
   * @param {string} token - The token of the Telegram bot.
   * @param {string} chatId - The chat ID of the Telegram group.
   */
  constructor(token, chatId) {
    this.bot = new TelegramBot(token, { polling: true });
    this.chatId = chatId;
  }

  /**
   * Sends a message to the Telegram group.
   *
   * @param {string} message - The message to be sent to the group.
   * @returns {Promise<Object>} - The response from the Telegram API.
   * @throws {Error} - If there is an error sending the message.
   */
  async sendMessageToGroup(message) {
    try {
      const response = await this.bot.sendMessage(this.chatId, message);
      console.log("Message sent successfully:", response);
      return response;
    } catch (error) {
      console.error("Error sending message:", error);
      throw error;
    }
  }

  /**
   * Stops the Telegram bot from polling for new messages.
   */
  stopBot() {
    this.bot.stopPolling();
    console.log("TelegramSender Bot stopped.");
  }
}

module.exports = TelegramManager;
